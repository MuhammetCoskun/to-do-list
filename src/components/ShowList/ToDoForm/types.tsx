export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
