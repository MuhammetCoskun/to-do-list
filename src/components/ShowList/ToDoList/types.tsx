import { ListType } from "../../MyLists/Lists/types";

export type ToDoListItemType = {
  id: string;
  text: string;
  createdAt: string;
  isDone: boolean;
  doneAt?: string;
};

export type ToDoListItemProps = {
  id: string;
  key?: string;
  text: string;
  onRemove: (id: string) => void;
};

export type ToDoListProps = {
  list: ListType;
  onRemove: (id: string) => void;
};
