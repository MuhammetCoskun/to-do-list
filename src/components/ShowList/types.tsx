import { ListType } from "../MyLists/Lists/types";

export type ShowListProps = {
  listToShow: ListType;
  onAddTodo: (text: string) => void;
  onRemoveTodo: (id: string) => void;
};
