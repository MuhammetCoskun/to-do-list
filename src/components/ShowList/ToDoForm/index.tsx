import React, { useRef, useState } from "react";
import { ToDoFormProps } from "./types";

const ToDoForm: React.FC<ToDoFormProps> = ({ onAddTodo }) => {
  const todoRef = useRef<HTMLInputElement>(null);
  const [inputIsValid, setInputIsValid] = useState<boolean>(true);

  const submitHandler = (event: React.FormEvent) => {
    event.preventDefault();
    const enteredTodo = todoRef.current!.value;
    if (enteredTodo?.trim().length === 0) {
      setInputIsValid(false);
      return;
    }
    setInputIsValid(true);
    onAddTodo(enteredTodo);
    todoRef.current!.value = "";
  };

  return (
    <div className={"task-form"}>
      <form onSubmit={submitHandler}>
        <input
          className="new todo"
          type="text"
          placeholder="Enter a new to do"
          aria-label="new to do name"
          ref={todoRef}
          data-testid="input"
        />
        <button
          aria-label="create new to do"
          data-testid="button"
          className="btn create"
        >
          +
        </button>
      </form>
      <div className="delete-item-container">
        <button className="btn delete">Clear completed tasks!</button>
        <button className="btn delete">Delete list!</button>
      </div>
      {!inputIsValid && (
        <p data-testid="paragraph">Please enter a valid to do..</p>
      )}
    </div>
  );
};

export default ToDoForm;
//TODO add style
