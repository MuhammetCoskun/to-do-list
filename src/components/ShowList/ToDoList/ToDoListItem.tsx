import React from "react";
import { ToDoListItemProps } from "./types";

const ToDoListItem: React.FC<ToDoListItemProps> = ({
  id,
  text,
  isDone,
  onChangeIsDone,
}) => {
  const clickHandler = () => {
    onChangeIsDone(id);
  };

  return (
    <div className={"todo-task-item"}>
      <input type="checkbox" id={id} defaultChecked={isDone} />
      <label htmlFor={id} onClick={clickHandler}>
        <span className={"checkbox"}></span>
        {text}
      </label>
    </div>
  );
};

export default ToDoListItem;
