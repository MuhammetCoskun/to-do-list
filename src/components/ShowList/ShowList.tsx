import ToDoForm from "./ToDoForm";
import ToDoList from "./ToDoList";
import { ShowListProps } from "./types";

const ShowList: React.FC<ShowListProps> = ({
  listToShow,
  onAddTodo,
  onRemoveTodo,
  onRemoveList,
  onChangeIsDone,
}) => {
  return (
    <div className="todo-list-show">
      <div className={"list-header"}>
        <h2 className={"list-title"}>{listToShow.title}</h2>
        {listToShow.todos.length > 0 && (
          <p className="todo-count">
            {listToShow.todos.length} tasks remaining!
          </p>
        )}
      </div>
      <div className="todo-body">
        <ToDoList list={listToShow} onChangeIsDone={onChangeIsDone} />
        <ToDoForm
          onAddTodo={onAddTodo}
          onRemoveTodo={onRemoveTodo}
          onRemoveList={onRemoveList}
        />
      </div>
    </div>
  );
};

export default ShowList;
