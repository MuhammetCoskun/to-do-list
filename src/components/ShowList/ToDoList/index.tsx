import React from "react";
import ToDoListItem from "./ToDoListItem";
import { ToDoListProps } from "./types";

const ToDoList: React.FC<ToDoListProps> = ({ list, onRemove }) => {
  return (
    <div className={"todo-tasks-container"}>
      {list.todos.map((item) => (
        <ToDoListItem
          key={item.id}
          id={item.id}
          text={item.text}
          onRemove={onRemove}
        />
      ))}
    </div>
  );
};

export default ToDoList;
