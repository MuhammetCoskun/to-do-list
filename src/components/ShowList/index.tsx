import ToDoForm from "./ToDoForm";
import ToDoList from "./ToDoList";
import { ShowListProps } from "./types";

const ShowList: React.FC<ShowListProps> = ({
  listToShow,
  onAddTodo,
  onRemoveTodo,
}) => {
  return (
    <div className="todo-list-show">
      <div className={"list-header"}>
        <h2 className={"list-title"}>{listToShow.title}</h2>
        {listToShow.todos.length > 0 && (
          <p className="todo-count">
            {listToShow.todos.length} tasks remaining!
          </p>
        )}
      </div>
      <div className="todo-body">
        <ToDoList list={listToShow} onRemove={onRemoveTodo} />
        <ToDoForm onAddTodo={onAddTodo} />
      </div>
    </div>
  );
};

export default ShowList;
