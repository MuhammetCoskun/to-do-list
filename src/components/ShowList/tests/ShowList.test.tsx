import React from "react";
import { render } from "@testing-library/react";
import ShowList from "../ShowList";
import { ToDoListProps } from "../ToDoList/types";

const mockOnAddTodo = jest.fn();
const mockOnRemoveTodo = jest.fn();
const mockOnRemoveList = jest.fn();
const mockOnChangeIsDone = jest.fn();

jest.mock(
  "../ToDoList/ToDoList",
  () =>
    ({ list, onChangeIsDone }: ToDoListProps) =>
      (
        <div id={list.id}>
          {list.todos.map((todo) => (
            <p key={todo.id}>{todo.text}</p>
          ))}
        </div>
      )
);

jest.mock("../ToDoForm/ToDoForm", () => () => (
  <div>Form Input and Buttons</div>
));

describe("ShowList component", () => {
  test("should render correctly", () => {
    const { asFragment } = render(
      <ShowList
        onAddTodo={mockOnAddTodo}
        onRemoveTodo={mockOnRemoveTodo}
        onChangeIsDone={mockOnChangeIsDone}
        onRemoveList={mockOnRemoveList}
        listToShow={{
          id: "1",
          title: "title",
          todos: [
            { id: "2", text: "text1", createdAt: "2022", isDone: false },
            { id: "3", text: "text2", createdAt: "2022", isDone: false },
          ],
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
