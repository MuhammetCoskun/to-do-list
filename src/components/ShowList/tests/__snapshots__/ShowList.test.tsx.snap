// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowList component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="todo-list-show"
  >
    <div
      class="list-header"
    >
      <h2
        class="list-title"
      >
        title
      </h2>
      <p
        class="todo-count"
      >
        2 tasks remaining!
      </p>
    </div>
    <div
      class="todo-body"
    >
      <div
        id="1"
      >
        <p>
          text1
        </p>
        <p>
          text2
        </p>
      </div>
      <div>
        Form Input and Buttons
      </div>
    </div>
  </div>
</DocumentFragment>
`;
